cmds/dhclient/dhclient.go

```diff
@@ -205,26 +207,36 @@ func dhclient4(iface netlink.Link, numRenewals int, timeout time.Duration, retry
 // dhcp6 support in go is hard to find. This function represents our best current
 // guess based on reading and testing.
 func dhclient6(iface netlink.Link, numRenewals int, timeout time.Duration, retry int) error {
-	conn, err := dhcp6.NewPacketSock(iface.Attrs().Index)
+	client, err := dhcp6client.New(iface, timeout, retry)
 	if err != nil {
-		return fmt.Errorf("client connection generation: %v", err)
+		return err
 	}
-	client := dhcp6.New(iface.Attrs().HardwareAddr, conn, timeout, retry)
 
 	for i := 0; numRenewals < 0 || i < numRenewals+1; i++ {
 		debug("Start getting or renewing DHCPv6 lease")
-		iaAddrs, packet, err := client.Solicit()
+		packet, err := client.Solicit()
 		if err != nil {
 			return fmt.Errorf("error: %v", err)
 		}
 		debug("Packet: %+v\n\n", packet)
+		// get IA_NA
+		iana, err := dhcp6opts.GetIANA(packet.Options)
+		if err != nil {
+			return err
+		}
+		// get IAAddrs from IA_NA
+		iaAddrs, err := dhcp6opts.GetIAAddr(iana[0].Options)
+		if err != nil {
+			return err
+		}
 		debug("IAAddrs: %v\n", iaAddrs)
 
 		if *test == false {
+			mask := net.IPMask(net.ParseIP("ffff:ffff:ffff:ffff::"))
```

Comment:
```go
mask := net.IPMask(bytes.Repeat([]byte{0xff}, net.IPv6len))
```

Reply:
this is actually a /64, so should become

```
mask := net.IPMask(bytes.Repeat([]byte{0xff}, net.IPv6len / 2))
```

Reply:
or better explicitly 8, instead of `net.IPv6len / 2`, plus the remaining 8 zero'ed bytes

Reply:
In other words it would become

```go
       mask := net.IPMask(append(
               bytes.Repeat([]byte{0xff}, 8),
               bytes.Repeat([]byte{0x00}, 8)...,
       ))
```
which is more verbose than using `net.ParseIP`. For now I'm committing just the other fix. Let me know what to do with this one